src/main/java/com/jhuguet/sb_taskv1/app/services/authorize/RequestAuthorization.java

```diff
@@ -0,0 +1,44 @@
+package com.jhuguet.sb_taskv1.app.services.authorize;
+
+import com.jhuguet.sb_taskv1.app.exceptions.UnqualifiedAuthority;
+import com.jhuguet.sb_taskv1.app.services.impl.CustomUserDetailsServiceImpl;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.Key;
+
+public class RequestAuthorization {
```

Comment:
I suggest you to parse jwt token in  in some filter. As a result, after parsing, you will create object UsernamePasswordAuthenticationToken (as I see the logic is already implemented in CustomAuthenticationProvider) and put it to SecurityContext

Reply:
But then, using principal as a resource to get current session information, when would we use the JWT?

Reply:
Jwt is validated in filter. And then you need to build Authentication object (UsernamePasswordAuthenticationToken) from token. And this object is used as a principal in controller.